bootstraptest/test_yjit_new_backend.rb

```diff
@@ -448,6 +448,14 @@ def bar
     [Foo.new.foo, bar]
 }
 
+# setinstancevariable
+assert_equal 'foo', %q{
+  def foo
+    @foo = "foo"
+  end
+  foo
+}
```

Comment:
It would be good to have at least one more test with something like 4 ivar slots

Reply:
Added a couple more test cases. https://github.com/Shopify/ruby/pull/362/commits/3f4c2439c50117237243d731e2df0b0e461134e2